htdocs/contrat/services_list.php

```diff
@@ -412,18 +412,28 @@
 if (!empty($filter_opcloture) && $filter_opcloture != -1) {
 	$param .= '&amp;filter_opcloture='.urlencode($filter_opcloture);
 }
+//removing PHP warnings
+$filter_dateouvertureprevue = isset($filter_dateouvertureprevue) ? $filter_dateouvertureprevue : '' ;
```

Comment:
A better fix would be to make
$filter_dateouvertureprevue = '';
$filter_date1 = '';
$filter_date2 = '';
$filter_opcloture = '';

at line 293 (so just before the code that may change the value to something else).

Reply:
Wouldn't that overwrite the variables if they were defined in including or included files?

Reply:
You must know one thing: 
Into a main page, the only var that are (shoud be) defined into external include files are the main global var:
$mysoc (object that store your company), 
$conf (storing all the setup), 
$user (the current logged user), 
$hookmanager (a utility class), 
$lang (the language to use for output). 
So if var is not this one, it is missing an init.   Only *.tpl.php file can use other vars that are not initialized because they are template file included into main pages.

Reply:
Cool, thanks :)